### DISCUSSION TITLE
HTML dialog does not render correctly on hover in Time Travel window

### DISCUSSION THREAD

**Discussion by KentDi801** [cypress-io/cypress#29293] [Questions and Help]
Our application modals have recently been converted from using css and divs to using the native HTML dialog element. If you run the following test you will see that the dialog appears as a modal dialog with a scrim visible. If you hover over the steps in the `time travel` window you will see that the dialog is displayed modelessly. This is causing issues when sending snapshots to Applitools Eyes.

```javascript
describe("Cypress Dialog tests", () => {
  it.only("should display react modal", () => {
    cy.visit("https://2g4q4s.csb.app?standalone");

    cy.get("[data-testid='btnTest']").click();
  });
});
```
Thanks in advance for any assistance.

**Comment by portyu9** [comment]
Hi @KentDi801 

This is consistent with a limitation of Cypress's Time Travel snapshot rather than the live `<dialog>` behaving incorrectly.

`dialog.showModal()` does more than add the `open` state to the DOM. The browser also places the dialog in its internal **top layer** and creates the `::backdrop`.

When Cypress restores an earlier DOM snapshot while you hover a Command Log entry, it can restore the captured DOM state, but browser-managed state such as top-layer membership is not simply represented by the HTML snapshot. That can make a modal `<dialog>` appear modeless when viewing the historical snapshot.

So I would not use the hovered Time Travel representation as the source for a visual assertion of a native modal dialog. Capture the visual assertion while the application is in its **live current state**, immediately after opening and asserting the dialog, for example:

```js
cy.get('[data-testid="btnTest"]').click()

cy.get('dialog').should(($dialog) => {
  expect($dialog[0].open).to.be.true
})

// Run the Applitools visual check here, against the current app state.
```

If Applitools still loses the backdrop/top-layer state when capturing the live page, that would be a separate visual-rendering integration issue to investigate with the Eyes SDK.

Please mark this answer as accepted if it helped, thank you.
